crates/meilisearch/src/routes/multi_search.rs

```diff
@@ -202,13 +202,20 @@ pub async fn multi_search_with_post(
                 .headers()
                 .get(PROXY_SEARCH_HEADER)
                 .is_some_and(|value| value.as_bytes() == PROXY_SEARCH_HEADER_VALUE.as_bytes());
+            let include_metadata = req
+                .headers()
+                .get(INCLUDE_METADATA_HEADER)
+                .and_then(|h| h.to_str().ok())
+                .map(|v| matches!(v.to_lowercase().as_str(), "true" | "1"))
+                .unwrap_or(false);
```

Comment:
this, and line 238-244 can be computed once before the match as it doesn't depend on whether we're federating or not.